watertap/unit_models/tests/test_nanofiltration_DSPMDE_0D.py

```diff
@@ -69,21 +70,23 @@
 def test_config():
     m = ConcreteModel()
     m.fs = FlowsheetBlock(default={"dynamic": False})
-    # NOTE: Not giving actual charges in order to test construction of
-    #   both the ion_set and solute_set
     m.fs.properties = DSPMDEParameterBlock(
-        default={"solute_list": ["Ca_2+", "SO4_2-", "Na_+", "Cl_-", "Mg_2+"]}
+        default={
+            "solute_list": ["Ca_2+", "SO4_2-", "Na_+", "Cl_-", "Mg_2+"],
+            "charge": {"Ca_2+": 2, "SO4_2-": -2, "Na_+": 1, "Cl_-": -1, "Mg_2+": 2},
+        }
```

Comment:
So right now you are only set up for this 1 specific test case, but given the number of issues this model has had, I think we need to add in some additional tests.

(1) Unit tests with and without Concentration Polarization
(2) Unit tests with the different Mass Transfer Coefficient options
(3) Unit tests with different sets of ions (this set + an NaCl set)
(4) Unit tests with different inlet conditions (i.e., stress testing)
(5) Unit tests with fixed outlets instead of inlets (i.e., an inverse solve to show model solution is truly unique)

Reply:
I totally agree. I'll check off what I can, and maybe we can revisit to cast judgement on what should be in this PR and what can be swept to a subsequent PR.

- [ ] Unit tests with and without Concentration Polarization
- [ ] Unit tests with the different Mass Transfer Coefficient options
- [ ] Unit tests with different sets of ions (this set + an NaCl set)
- [ ] Unit tests with different inlet conditions (i.e., stress testing)
- [ ] Unit tests with fixed outlets instead of inlets (i.e., an inverse solve to show model solution is truly unique)